comment/mixins.py

```diff
@@ -1,16 +1,40 @@
 import abc
 
 from django.contrib.auth.mixins import LoginRequiredMixin, AccessMixin
-from django.contrib import messages
 from django.core.exceptions import ImproperlyConfigured
-from django.http import HttpResponseBadRequest, HttpResponseForbidden
+from django.http import HttpResponseBadRequest, HttpResponseForbidden, JsonResponse
+from django.views.generic import FormView
 
 from comment.conf import settings
 from comment.utils import is_comment_admin, is_comment_moderator
 from comment.validators import ValidatorMixin
 from comment.messages import ErrorMessage, FlagError, FollowError
 from comment.service.email import DABEmailService
 from comment.messages import EmailInfo
+from comment.forms import CommentForm
+from comment.context import DABContext
+
+
+class DABResponseData:
+    status = 200
+    data = None
+    error = None
+    anonymous = False
+    msg = None
+
+    def json(self):
+        return {
+            'data': self.data,
+            'error': self.error,
+            'anonymous': self.anonymous,
+            'msg': self.msg
+        }
+
+
+class UTF8JsonResponse(JsonResponse):
+    def __init__(self, *args, json_dumps_params=None, **kwargs):
+        json_dumps_params = {"ensure_ascii": False, **(json_dumps_params or {})}
+        super().__init__(*args, json_dumps_params=json_dumps_params, **kwargs)
```

Comment:
maybe we can move these to a separate module `responses.py`.

Reply:
Seems eligible.